MailManager/src/tests/java/inc/softserve/EnvelopeServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package inc.softserve;
+
+import inc.softserve.common.EnvelopeTools;
+import inc.softserve.dao.EnvelopeDao;
+import inc.softserve.dao.UserDao;
+import inc.softserve.services.EnvelopeService;
+import inc.softserve.services.EnvelopeServiceImpl;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class EnvelopeServiceTest extends Assert {
+
+    EnvelopeService service;
+
+    @BeforeClass
+    public void setUpDataSource() throws Exception {
+
+        UserDao usrDao = mock(UserDao.class);
+        User user = new User("akonon", "a", "b");
+        when(usrDao.getUserByUsername(anyString())).thenReturn(null);
+        when(usrDao.getUserByUsername("akonon")).thenReturn(user);
+
+        Envelope envelopeNotFound = new Envelope(user, "akonon22@mail.ru", "TestEmail", "Some template", EnvelopeState.NOT_FOUND);
```

Comment:
Please use a bit more abstract data in tests. test@example.com would be a safer bet.